server/src/services/metadata.service.ts

```diff
@@ -487,6 +496,124 @@ export class MetadataService implements OnEvents {
     }
   }
 
+  async searchPerson(ownerId: AssetEntity['ownerId'], dto: SearchPeopleDto): Promise<PersonResponseDto[]> {
+    return this.personRepository.getByName(ownerId, dto.name, { withHidden: dto.withHidden });
+  }
+
+  private async applyTaggedFaces(asset: AssetEntity, tags: ImmichTags) {
+    this.logger.debug(`Starting faces extraction from tags (${asset.id})`);
+
+    const faceSet = new Set();
+
+    if (!tags.RegionInfo) {
+      return true;
+    }
+
+    interface Region {
+      imageWidth: number;
+      imageHeight: number;
+      boundingBoxX1: number;
+      boundingBoxX2: number;
+      boundingBoxY1: number;
+      boundingBoxY2: number;
+    }
+
+    interface MetadataFaceResult {
+      Name?: string;
+      Type?: string;
+      Region: Region;
+    }
+
+    function toPixels(region: Region, unit: string): Region {
+      const pixelRegion: Region = region;
+      if (unit.toLowerCase() == 'normalized') {
+        pixelRegion.boundingBoxX1 = Math.floor(region.boundingBoxX1 * region.imageWidth);
+        pixelRegion.boundingBoxX2 = Math.floor(region.boundingBoxX2 * region.imageWidth);
+        pixelRegion.boundingBoxY1 = Math.floor(region.boundingBoxY1 * region.imageHeight);
+        pixelRegion.boundingBoxY2 = Math.floor(region.boundingBoxY2 * region.imageHeight);
+      }
+      return pixelRegion;
+    }
```

Comment:
```suggestion
    const toPixels = (region: Region, unit: string): Region => {
      if (unit.toLowerCase() === 'normalized') {
        return {
          boundingBoxX1: Math.floor(region.boundingBoxX1 * region.imageWidth),
          boundingBoxX2: Math.floor(region.boundingBoxX2 * region.imageWidth).
          boundingBoxY1: Math.floor(region.boundingBoxY1 * region.imageHeight),
          boundingBoxY2: Math.floor(region.boundingBoxY2 * region.imageHeight),
        }
      }
      
      return region;
    }
```

Reply:
Or alternatively, this could be a private method in the service too. I think I'd prefer that.

Reply:
Agreed, moved!